src/ghgvc_script.R

```diff
@@ -2,185 +2,94 @@
-plotdata[is.na(plotdata)] <- 0
-
-plotdata$CRV_BGC[plotdata$CRV_NET == 0] <- 0
-plotdata$CRV_BIOPHYS[plotdata$CRV_NET == 0] <- 0
-plotdata$CRV_NET[plotdata$CRV_NET == 0] <- NA
-
-nolabels <- theme(axis.title = element_blank(), axis.text.y = element_blank(), 
-                  axis.ticks.y = element_blank(), legend.position = "top")
-baseplot <- ggplot(data = plotdata) + theme_minimal() + coord_flip() + nolabels + theme(panel.grid.major = element_line("black", size = 0.14)) +  geom_hline(aes(yintercept =0)) 
-
-
-## T_E is number of years considered
-YEARS <- config.list$options$T_E
-xlabels <- as.expression(bquote(paste("CO"[2], " Emission Equivalents (Mg CO"[2],"-eq ha"^{-1}, " ", .(YEARS)," yrs"^{-1},")")))
-
-biome <- data.frame(order = 1:(nrow(plotdata)+1), Biome = c("", as.character(plotdata$Biome)))
-
-longdata <- melt(plotdata, id.var = "Biome")
-longdata$label <- gsub(" Site", "\nSite", longdata$Biome)
-
-
-ghgvc.subplot <- function(vars, data = longdata){
-  d <- subset(longdata, variable %in% vars)
-  pos <- d$value > 0
-  posplot <- if(any(pos)) {
-    geom_bar(data = d[pos,], 
-             aes(x = Biome, y = value, fill = variable),  
-             width = 0.25, stat = "identity")  
-  } else {
-    NULL
-  }
-  negplot <- if(any(!pos)) {
-    geom_bar(data = d[!pos,], 
-             aes(x = Biome, y = value, fill = variable),  
-             width = 0.25, stat = "identity")  
-  } else {
-    NULL
-  }
-  return(baseplot + posplot + negplot)
-}
-
-bgc.plot <- ghgvc.subplot(c("Storage", "Ongoing_Exchange"), data = longdata) +
-  scale_fill_manual(values= brewer_pal(pal = "Greens")(6)[c(4,6)], labels = c("Storage", "Ongoing Exchange")) + labs(fill = "") +
-  ggtitle("Biogeochemical") + theme(axis.text.y = element_text(size = 12, hjust = 1))
-
-biophys.plot <- ghgvc.subplot(c("LE", "Rnet"), data = longdata) + 
-  scale_fill_manual(values = brewer_pal(pal = "Blues")(6)[c(4,6)], labels = c(expression("LE", "R"["net"]))) + labs(fill = "") + 
-  ggtitle("Biophysical")
-
-crv.plot <-  ghgvc.subplot(c("CRV_BGC", "CRV_BIOPHYS"), data = longdata) +
-  scale_fill_manual(values= c(brewer_pal(pal = "Greens")(6)[5], brewer_pal(pal = "Blues")(6)[5]), labels = c("Biogeochemical", "Biophysical")) + labs(fill = "") +
-  geom_point(data = subset(longdata, variable == "CRV_NET"), aes(x = Biome, y = value)) +
-  ggtitle("Climate Regulating Value")
-
-svg(filename=file.path(outdir, "output.svg"), width = 10, height = 1 + nrow(plotdata))
-
-grid.arrange(bgc.plot, biophys.plot, crv.plot, ncol = 3, widths = c(2,1,1),
-             sub = textGrob(xlabels, hjust = 0.2))
-dev.off()
-
-svgfixcmd <- paste("sed -i 's/symbol id/symbol overflow=\"visible\" id/g'", file.path(outdir, "output.svg"))
-system(svgfixcmd)
-#sub = annotate("text", x = 2, y = 0.3, parse = T, label = xlabels)))
+### Write the results to a file
+ghgcv_write(x, outdir, format = c("json", "csv"))
+
+
+### Create a plot of the results
+p = ghgcv_plot(x, outdir, config.list$options$T_E)
```

Comment:
replace `=` with `<-` and, if you haven't done so yet, please review Hadley's style guide.